### PR TITLE
Instance config editor

### PR DESCRIPTION
**Pull request by d-loose**
~~WIP~~
- show instance configuration in info dialog (hide `volatile.*`, `user.workshops.*`, `image.*`
- add instance config schema, without wildcards for now

Todo:
- [x] handle `blob` config entry type
- [x] handle multiline settings `security.syscalls.allow` and `security.syscalls.deny`
- [x] handle async errors
image
image

### CODE REVIEW

File: lib/config_editor/config_editor_dialog.dart [hunk 1]
```diff
@@ -82,7 +82,34 @@ class ConfigEditor extends StatelessWidget {
           ),
         );
         break;
-      case bool:
+      case 'blob':
+      case 'multiline string':
+        final controller = ScrollController();
```
Comment (jpnurmi) [inline lib/config_editor/config_editor_dialog.dart:87]: the controller is not disposed, but is the controller actually needed?

File: lib/config_editor/config_editor_dialog.dart [hunk 2]
```diff
@@ -82,7 +82,34 @@ class ConfigEditor extends StatelessWidget {
...
+        child = SizedBox(
+          width: 200,
+          child: Scrollbar(
```
Comment (jpnurmi) [inline lib/config_editor/config_editor_dialog.dart:90]: doesn't SingleChildScrollView create a scrollbar automatically on desktop?

**Review by jpnurmi — COMMENTED** [review]
Very nice!

File: lib/config_editor/config_editor_dialog.dart (hunk 2 above)
Comment (d-loose) [reply lib/config_editor/config_editor_dialog.dart:90]: I don't get a scrollbar without the `Scrollbar` widget. Also, if I use it without the controller it's visible but doesn't work and flutter complains:
```
The Scrollbar's ScrollController has no ScrollPosition attached.
```

Comment (jpnurmi) [reply lib/config_editor/config_editor_dialog.dart:90]: Ok, nevermind then. Maybe it has something to do with these nested scroll views. :eyes:

File: lib/config_editor/config_editor_dialog.dart (hunk 1 above)
Comment (jpnurmi) [reply lib/config_editor/config_editor_dialog.dart:87]: ok, got confirmed that it's needed so we just need to find a way to dispose it after use

Comment (d-loose) [reply lib/config_editor/config_editor_dialog.dart]: Not sure if there's a better way of doing this - I've added a stateful widget to handle this now

File: lib/config_editor/config_editor_dialog.dart [hunk 3]
```diff
@@ -178,6 +188,52 @@ class ConfigEditor extends StatelessWidget {
...
+}
+
+class __MultiLineTextFieldState extends State<_MultiLineTextField> {
+  late final ScrollController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = ScrollController();
+  }
```
Comment (jpnurmi) [inline lib/config_editor/config_editor_dialog.dart:207]: cosmetic nit-pick: we can simplify this to
```dart
final _controller = ScrollController();
```dart

we don't need `initState()` at all when there's no need to access `widget` for initial values, add listeners, or things like that

**Review by jpnurmi — APPROVED** [review]
LGTM :+1:

**Pull request merged**